File: Projects/06/quad.java
import java.lang.Math;
import java.util.Scanner;

public class quad {
    public static boolean quadraticRoots(double a, double b, double c,
                                         double[] roots) {
        if (a != 0) {
            double arg = Math.pow(b, 2.0) - 4 * a * c;
            if (arg >= 0) {
                roots[0] = (-b + Math.sqrt(arg)) / (2 * a);
                roots[1] = (-b - Math.sqrt(arg)) / (2 * a);
                return true;
            } else {
                System.err.println("\n*** quadraticRoots(): b^2 - 4ac is negative!");
                return false;
            }
        } else {
            System.err.println("\n*** QuadraticRoots(): a is zero!");
            return false;
        }
    }

    public static void main(String[] args) {
        double a, b, c;
        double[] roots = new double[2];

        System.out.println("Enter the values of a, b, and c:");
        Scanner scanner = new Scanner(System.in);

        a = scanner.nextDouble();
        b = scanner.nextDouble();
        c = scanner.nextDouble();

        if (quadraticRoots(a, b, c, roots)) {
            System.out.println("The roots are: " + roots[0] + " and " + roots[1]);
        }
    }
}

File: Projects/06/quad.rb
def quadratic_roots(a, b, c)
    if a != 0
      arg = b**2 - 4*a*c
      if arg >= 0
        root1 = (-b + Math.sqrt(arg))/(2*a)
        root2 = (-b - Math.sqrt(arg))/(2*a)
        return [root1, root2]
      else
        puts "\n*** quadratic_roots: b^2 - 4ac is negative!"
        return [0, 0]
      end
    else
      puts "\n*** quadratic_roots: a is zero!"
      return [0, 0]
    end
  end

puts "Enter a: "
a = gets.chomp.to_f
puts "Enter b: "
b = gets.chomp.to_f
puts "Enter c: "
c = gets.chomp.to_f

roots = quadratic_roots(a, b, c)

if roots != [0, 0]
  puts "Root 1: #{roots[0]}"
  puts "Root 2: #{roots[1]}"
else
  puts "Invalid roots."
end